packages/vite/src/node/plugins/css.ts

```diff
@@ -919,17 +919,15 @@ async function doUrlReplace(
   matched: string,
   replacer: CssUrlReplacer
 ) {
-  let wrap = ''
   const first = rawUrl[0]
-  if (first === `"` || first === `'`) {
-    wrap = first
-    rawUrl = rawUrl.slice(1, -1)
-  }
-  if (isExternalUrl(rawUrl) || isDataUrl(rawUrl) || rawUrl.startsWith('#')) {
+  const [wrap, url] =
+    first === `"` || first === `'` ? [first, rawUrl.slice(1, -1)] : ['', rawUrl]
```

Comment:
@bluwy 

Though it's using destructuring, it is common usage to declare multiple related variables.

I personally prefer the style `const [width, height] = [100, 200]` over `const width = 100; const height = 200;` for readbility reason.

Maybe we needn't think about the engine's stuff. And we didn't consist the coding style everywhere. e.g. https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/importAnalysis.ts#L394-L398

Reply:
I personally find multiple variables more readable 😬 The code creates and then destructures the array, which allocates memory and affects performance. If we can avoid that to squeeze performance, especially for a small, contained code like this, I think it's a big plus in the long run.

The code you linked uses destructuring because there were multiple values returned from the function. The array is already allocated in memory.

Reply:
Yeah from the performance perspective, you are right but it is almost an endless road. A tradeoff principle is considering whether it is space performance-sensitive but I'm not sure the `doUrlReplace` is.
This PR is asking for adding functional style with a slight performance sacrifice.

Reply:
I agree with @bluwy here, thanks for the PR @zheeeng, I think we should keep the code as is for now.